server/rpc.go

```diff
@@ -297,6 +331,21 @@ func getOrchestrator(orch Orchestrator, req *net.OrchestratorRequest) (*net.Orch
 	return orchestratorInfo(orch, addr, orch.ServiceURI().String())
 }
 
+func endSession(node *core.LivepeerNode, orch Orchestrator, req *net.EndSessionRequest) (*net.EndSessionResponse, error) {
+	addr := ethcommon.BytesToAddress(req.Address)
+	if err := verifyOrchestratorReq(orch, addr, req.Sig); err != nil {
+		return nil, fmt.Errorf("Invalid orchestrator request: %v", err)
+	}
+
+	if _, err := authenticateBroadcaster(addr.Hex()); err != nil {
+		return nil, fmt.Errorf("authentication failed: %v", err)
+	}
+
+	node.TranscoderManager.EndSession(req.SessionId)
```

Comment:
Seems like we would want to execute all the cleanup logic that we would otherwise run when we hit a timeout in the O's [transcode loop](https://github.com/livepeer/go-livepeer/blob/e3f9286f80044dc1e55a4687eaeccb689a2ebf84/core/orchestrator.go#L638) here as well - perhaps we could extract all of that logic into a shared function that is triggered in the following circumstances:

- If the O's transcode loop times out due to inactivity (we already do this)
- If the O receives an EndSessionRequest from B